hvega/CHANGELOG.md

```diff
@@ -1,6 +1,150 @@
 For the latest version of this document, please see
 [https://github.com/DougBurke/hvega/blob/master/hvega/CHANGELOG.md](https://github.com/DougBurke/hvega/blob/master/hvega/CHANGELOG.md).
 
+## 0.5.0.0
+
+Update to version 4.0 of the Vega-Lite specification (tested against
+version 4.0.2). There are several changes in default behavior due
+to this (tooltips are now disabled by default, the background now defaults
+to white rather than transparent), although this is also controlled by
+how the Vega-Lite visualization is rendered, which can make tracking
+down why something has changed a bit awkward.
+
+There is more-extensive use of type aliases, such as 'Color',
+and the introduction of several more (e.g. 'DashStyle' and `FieldName`).
+These do not add any type safety, but help the documentation (as they provide
+a single place to explain the meaning and any constraints on a
+particular value).
+
+Minor documentation improvements.
+
+### New functionality
+
+Title (and subtitle) strings can now be split across multiple lines:
+use '\n' to indicate a line break.
+
+Colors are now stripped of extraneous white space, and if empty
+converted to the JSON null value rather than an empty string.
+
+The `pivot` transform has been added, along with the `PivotProperty`
+preferences type. This is the inverse of `fold`.
+
+The `density` transform has been added, along with the `DensityProperty`
+configuration type, to support kernel density estimation (e.g. to
+generate a continuous distribtion from a discrete one).
+
+The `loess` transform has been added, along with the `LoessProperty`
+configuration type, to support scatterplot smoothing.
+
+The `regression` transform has been added, along with the `RegressionProperty`
+and `RegressionMethod` configuration types, to support regression
+analysis.
+
+The `quantile` transform has been added, along with the `QuantileProperty`
+type, to support quantile analysis.
+
+The `url` encoding has been added, which allows you to view images (e.g.
+PNG) via the (new) `Image` mark type. The `MAspect` `MarkProperty` has
+been added as a configuration option.
+
+The `lookupSelection` transform has been added to support joining
+data via a selection. The `SelectionLabel` type alias has been added
+to help the documentation.
+
+The `heightOfContainer` and `widthOfContainer` functions
```

Comment:
These don't work with composed views unfortunately, perhaps worth mentioning.

Reply:
My one try at using them failed spectacularly, but I haven't got around to investigating why. Do you know if its a fundamental limitation of Vega Lite, or am I doing something wrong in @hvega@?

Reply:
I don't know, haven't tried them yet because I'm using composed views. It's disappointing that the API has all these non-compositional features, we're spoilt by Haskell libraries.

Reply:
Can you give an example or two of how you imagine the API could be compositional? At the moment hvega remains very-close to the Elm version, which means that it is very-much in the "Simple Haskell" camp. I'm interested to know if a more ergonomic API can be built (either on top of the existing one or replacing it), but am unsure if the underlying schema (and the fact it is currently chaning quite rapidly) is well suited to it.

Reply:
I don't have any suggestion for `hvega`. The problem is in vega-lite itself. You're doing an amazing job keeping up with the API changes.